src/runtime/ValueInlines.h

```diff
@@ -807,34 +807,7 @@ inline bool Value::toBoolean(ExecutionState& ec) const // $7.1.2 ToBoolean
     if (isInt32())
         return asInt32();
 
-    if (isDouble()) {
-        double d = asDouble();
-        if (std::isnan(d))
-            return false;
-        if (d == 0.0)
-            return false;
-        return true;
-    }
-
-    if (isUndefinedOrNull())
-        return false;
-
-    ASSERT(isPointerValue());
-
-    if (UNLIKELY(asPointerValue()->isString()))
-        return asString()->length();
-
-    if (UNLIKELY(isBigInt())) {
-        return !asBigInt()->isZero();
-    }
-
-#if defined(ESCARGOT_ENABLE_TEST)
-    if (UNLIKELY(checkIfObjectWithIsHTMLDDA())) {
-        return false;
-    }
-#endif
-    // Symbol, Objects..
-    return true;
+    return toBooleanSlowCase(ec);
```

Comment:
Q. How does this lead to performance improvement?

Reply:
the shorter Value::toBoolean can be inline-able to compilers.
the longer version of Value::toBoolean is not inlined by compilers.

Reply:
Thanks you for answer!
I remembered the godbolt you presented at the seminar last time in best reviewer course work.
so I tested it using this, I can see [only short function is inlined by gcc](https://godbolt.org/z/hY7bqPbM5)